src/render/painter.js

```diff
@@ -979,7 +986,27 @@ class Painter {
         if (fog) {
             const fogOpacity = fog.getOpacity(this.transform.pitch);
             if (fogOpacity !== 0.0) {
-                program.setFogUniformValues(context, fogUniformValues(this, fog, tileID, fogOpacity));
+                const tr = this.transform;
+                const viewMatrix = tr._camera.getWorldToCamera(tr.worldSize, 1.0);
+                const center = [tr.globeMatrix[12], tr.globeMatrix[13], tr.globeMatrix[14]];
+                const globeCenterInViewSpace = vec3.transformMat4(center, center, viewMatrix);
+                const globeRadius = tr.worldSize / 2.0 / Math.PI - 1.0;
+                const viewport = [
+                    tr.width * browser.devicePixelRatio,
+                    tr.height * browser.devicePixelRatio
+                ];
```

Comment:
Yes, there are at least two improvements to make there that I mentioned in the description: 
>Reduce frequency of globe specific uniforms when updating fog uniform, these values can be 1) calculated once per frame and 2) only be done when used along with globe

because these are evaluated for every draw call, and could be evaluated once per frame instead. I'll fix that in this PR.